app/assets/javascripts/host_power.js

```diff
@@ -0,0 +1,64 @@
+(function(){
+
+  $(document).ready(function () {
+    powerStatus();
+  });
+
+  function hostIds() {
+    var res = [];
+
+    $('#hosts-table tbody tr').each( function (index, item) {
+      res.push($(item).data("host-id"));
+    });
+
+    return res;
+  }
+
+  function findSpinner(id) {
+    return $('#hosts-table tbody').find("tr[data-host-id='" + id + "']").find(".spinner");
+  }
+
+  function updatePowerStatus(response) {
+    var spinner = null;
+    hostIds().forEach(function (item) {
+      spinner = findSpinner(item);
+      if ((typeof response !== "string") && response[item]) {
```

Comment:
[find_multiple](https://github.com/theforeman/foreman/blob/develop/app/controllers/hosts_controller.rb#L779) may render the entire hosts page.